### PR TITLE
Improve documentation and tests for raw_host_with_port and host_with_…

### PR DESCRIPTION
**Pull request by tomkadwill**
### Summary

Improve clarity of raw_host_with_port and host_with_port, with improved documentation and improved test coverage.

Currently it is not clear what the difference between raw_host_with_port and host_with_port is. This change makes it clearer, exactly what the differences are.

### CODE REVIEW

File: actionpack/lib/action_dispatch/http/url.rb [hunk 1]
```diff
@@ -217,7 +217,7 @@ def protocol
         @protocol ||= ssl? ? 'https://' : 'http://'
       end
 
-      # Returns the \host for this request, such as "example.com".
+      # Returns the \host with the port for this request, such as "example.com:8080".
```
Comment (maclover7) [inline actionpack/lib/action_dispatch/http/url.rb]: host and port

File: actionpack/lib/action_dispatch/http/url.rb [hunk 2]
```diff
@@ -249,12 +252,16 @@ def host
...
       # Returns a \host:\port string for this request, such as "example.com" or
-      # "example.com:8080".
+      # "example.com:8080". Port is only included if it is not a default port
+      # (80 or 443)
       #
       #   class Request < Rack::Request
       #     include ActionDispatch::Http::URL
       #   end
       #
+      #   req = Request.new 'HTTP_HOST' => 'example.com'
```
Comment (maclover7) [inline actionpack/lib/action_dispatch/http/url.rb]: what does this new example give us, that the next one doesn't? they look the same?

Comment (maclover7) [inline actionpack/lib/action_dispatch/http/url.rb]: ahh, never mind. saw the test you added 👍

File: actionpack/lib/action_dispatch/http/url.rb (hunk 1 above)
Comment (tomkadwill) [inline actionpack/lib/action_dispatch/http/url.rb]: Thanks for the review @maclover7, I've updated the PR.

**Comment by maclover7** [comment]
👍 from me, though I don't have permissions to merge anything that isn't solely documentation changes 😬

**Comment by kaspth** [comment]
Thanks!

**Pull request merged**